### PR TITLE
Database V2 Backend

### PR DESCRIPTION
**Pull request by ghost**
Backend for the database V2, contains all backend that reads from the database. Backend that adds to the database like 'Add', 'Allocate' etc still needs to be done

### CODE REVIEW

**Comment by JoshLmao** [comment]
Which database is meant to be used? We have 3 in there now and it's confusing. Can you delete old ones we're not using please? If we need to get them back, we can go back in the commits to get it

**Comment by ghost** [comment]
> Which database is meant to be used? We have 3 in there now and it's confusing. Can you delete old ones we're not using please? If we need to get them back, we can go back in the commits to get it

I've removed the old database files, it's called nhs_asset_tracker.sql 
7f2b0af

**Comment by JoshLmao** [comment]
Also, do you have data populated in your current database? It'd save me a lot of time if you could put that in too 😅
We could have the default db file (nhs_asset_tracker.sql)and another next to it with data populated (nhs_asset_tracker_populated.sql). If not, don't worry about it

File: api/assets/get/index.php [hunk 1]
```diff
@@ -38,7 +38,7 @@
 }
 
 /// Query to determine the next available asset id in table
-$sql = "SELECT * FROM assets WHERE id=" . $asset_id;
+$sql = "SELECT * FROM $ASSETS_TABLE WHERE Equi_ID=" . $asset_id;
```
Comment (JoshLmao) [inline api/assets/get/index.php]: Should the Equi_ID part of this come from the api_config variable?

File: src/components/Assets/SingleAsset/SingleAsset.jsx [hunk 2]
```diff
@@ -160,26 +161,31 @@ class SingleAsset extends Component
...
-            console.log(result);
-            this.setState({
-                asset: result.data.asset,
-                assetIdLoaded: true,
+        if ( Session.isSignedIn() ) {
+            /// Retrieve correct asset from database
+            axios({
+                method: 'get',
+                // No api set up currently, being left for now
+                url: `${BASE_API_PATH}/assets/get?id=${this.state.id}`,
```
Comment (JoshLmao) [inline src/components/Assets/SingleAsset/SingleAsset.jsx]: I think this should use the appropriate API function inside helperFile.js

File: src/components/EditUser/EditUser.jsx [hunk 3]
```diff
@@ -54,31 +54,33 @@ class EditUser extends Component {
...
-            });
-        }).catch(error => {
-            console.log(error);
-            this.setState({ 
-                error: error.message,
-                loaded: true, 
+        if ( Session.isSignedIn() ) {
+            axios({
+                method: 'GET',
+                url: `${BASE_API_PATH}/user/get?id=${this.state.id}`,
```
Comment (JoshLmao) [inline src/components/EditUser/EditUser.jsx]: Same here. Should use the API function in helperFile.js

File: src/components/Search/Search.jsx [hunk 4]
```diff
@@ -61,25 +62,31 @@ class Search extends Component {
...
-            timeout: API_TIMEOUT
-        }).then(result => {
-            this.setState({
-                results: result.data.assets,
-                loaded: true,
+        if ( Session.isSignedIn() ) {
+            /// Get relevent query results from api
+            axios({
+                method: 'get',
+                url: `${BASE_API_PATH}/assets/search?q=${this.state.query}`,
```
Comment (JoshLmao) [inline src/components/Search/Search.jsx]: Use the helperFile.js methods

**Review by JoshLmao — COMMENTED** [review]
Just change those url's to use the helperFile.js methods and it's good. You should be able to pass the data needed for those url's through as function parameter 😉

**Comment by ghost** [comment]
> Also, do you have data populated in your current database? It'd save me a lot of time if you could put that in too 😅
> We could have the default db file (nhs_asset_tracker.sql)and another next to it with data populated (nhs_asset_tracker_populated.sql). If not, don't worry about it

The database in the commit already has all the commits included within it. Anything with 'Insert Into' before it is just placing the dummy data into the table.

File: src/components/Search/Search.jsx (hunk 4 above)
Comment (ghost) [reply src/components/Search/Search.jsx]: > Just change those url's to use the helperFile.js methods and it's good. You should be able to pass the data needed for those url's through as function parameter 😉


4cfafd6 Added the changes you requested to the helperFile/frontend code

**Comment by JoshLmao** [comment]
> > Also, do you have data populated in your current database? It'd save me a lot of time if you could put that in too 😅
> > We could have the default db file (nhs_asset_tracker.sql)and another next to it with data populated (nhs_asset_tracker_populated.sql). If not, don't worry about it
> 
> The database in the commit already has all the commits included within it. Anything with 'Insert Into' before it is just placing the dummy data into the table.

Oh yeah, sorry. I imported it wrong and didn't notice you had dummy data already. Good job

**Comment by ghost** [comment]
> > > Also, do you have data populated in your current database? It'd save me a lot of time if you could put that in too 😅
> > > We could have the default db file (nhs_asset_tracker.sql)and another next to it with data populated (nhs_asset_tracker_populated.sql). If not, don't worry about it
> > 
> > 
> > The database in the commit already has all the commits included within it. Anything with 'Insert Into' before it is just placing the dummy data into the table.
> 
> Oh yeah, sorry. I imported it wrong and didn't notice you had dummy data already. Good job

No worries, I've done the exact same thing before with tables you've made

**Review by JoshLmao — APPROVED** [review]
All good. Thanks Matt. Merge whenever

**Pull request merged**